x-pack/legacy/plugins/siem/public/components/embeddables/actions/apply_siem_filter_action.test.tsx

```diff
@@ -179,6 +183,23 @@ describe('ApplySiemFilterAction', () => {
   });
 });
 
+describe('#getFilterExpression', () => {
+  test('it returns an empty expression if no filterValue is provided', () => {
+    const layerList = getFilterExpression('host.id', undefined);
+    expect(layerList).toEqual('(NOT host.id:*)');
+  });
+
+  test('it returns a valid expression when provided single filterValue', () => {
+    const layerList = getFilterExpression('host.id', 'aa8df15');
+    expect(layerList).toEqual('host.id: "aa8df15"');
+  });
+
+  test('it returns a valid expression when provided array filterValue', () => {
+    const layerList = getFilterExpression('host.id', ['xavier', 'angela', 'frank']);
+    expect(layerList).toEqual('(host.id: "xavier" OR host.id: "angela" OR host.id: "frank")');
+  });
```

Comment:
This test case works as well I tried out if you want to add it. An array with a single value:

```ts
  test('it returns a valid expression when provided array filterValue with a single value', () => {
    const layerList = getFilterExpression('host.id', ['xavier']);
    expect(layerList).toEqual('(host.id: "xavier")');
  });
```

Reply:
Thanks for the bounds test 🙂